### ISSUE TITLE
No creation esbmc file

### ISSUE THREAD

**Issue by vanderson-rocha** [esbmc/esbmc#274]
Hi team. I am following the step by step presented in https://github.com/esbmc/esbmc/blob/master/BUILDING.md, but in end the esbmc file is not being created. No error is displayed. Any sugestions?

**Comment by rafaelsamenezes** [comment]
Could you add more info?

1. Which OS.
2. The output of `cmake` command
3. The output of `cmake --build .` command

**Comment by vanderson-rocha** [comment]
Hi @rafaelsamenezes. Follow:

Which OS.
Ubuntu 20.04.1 LTS (GNU/Linux 5.4.0-47-generic x86_64)

The output of cmake command
-- sendfile found, esbmc_parseoptions will use it
-- ESBMC will be built in static mode
-- LIBM: /home/vanderson/mestrado/esbmc/src/c2goto/library/libm/*.c
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found version "1.71.0") found components: filesystem system date_time
-- LLVM version: 9.0.0
CMake Warning (dev) at scripts/cmake/OptimizationCCache.cmake:8 (message):
  ccache not found, incremental builds will be slower
Call Stack (most recent call first):
  CMakeLists.txt:53 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Multiarch include for c2goto: /usr/include/x86_64-linux-gnu. You may want to override C2GOTO_INCLUDE_DIR
-- Using Z3 at: /home/vanderson/mestrado/z3/bin/libz3.a
-- Z3 version: 4.8.6.0
CMake Warning (dev) at src/solvers/z3/CMakeLists.txt:55 (target_link_libraries):
  Policy CMP0079 is not set: target_link_libraries allows use with targets in
  other directories.  Run "cmake --help-policy CMP0079" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  Target

    solvers

  is not created in this directory.  For compatibility with older versions of
  CMake, link library

    solverz3

  will be looked up in the directory in which the target was created rather
  than in this calling directory.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Boolector at: /home/vanderson/mestrado/boolector-release/lib/cmake/Boolector
-- Boolector version: 3.2.0
CMake Warning (dev) at src/solvers/boolector/CMakeLists.txt:27 (target_link_libraries):
  Policy CMP0079 is not set: target_link_libraries allows use with targets in
  other directories.  Run "cmake --help-policy CMP0079" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  Target

    solvers

  is not created in this directory.  For compatibility with older versions of
  CMake, link library

    solverbtor

  will be looked up in the directory in which the target was created rather
  than in this calling directory.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found CVC4 at: /home/vanderson/mestrado/cvc4/lib/cmake/CVC4
-- CVC4 version: 1.8-prerelease
CMake Warning (dev) at src/solvers/cvc4/CMakeLists.txt:32 (target_link_libraries):
  Policy CMP0079 is not set: target_link_libraries allows use with targets in
  other directories.  Run "cmake --help-policy CMP0079" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  Target

    solvers

  is not created in this directory.  For compatibility with older versions of
  CMake, link library

    solvercvc4

  will be looked up in the directory in which the target was created rather
  than in this calling directory.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Using MathSAT at: /home/vanderson/mestrado/mathsat/lib/libmathsat.a
-- MathSAT version: 5.5.4
CMake Warning (dev) at src/solvers/mathsat/CMakeLists.txt:50 (target_link_libraries):
  Policy CMP0079 is not set: target_link_libraries allows use with targets in
  other directories.  Run "cmake --help-policy CMP0079" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  Target

    solvers

  is not created in this directory.  For compatibility with older versions of
  CMake, link library

    solvermsat

  will be looked up in the directory in which the target was created rather
  than in this calling directory.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Using Yices at: /home/vanderson/mestrado/yices/lib/libyices.a
-- Yices version: 2.6.1
-- Danger Will Robinson: the version of Yices you have is GPL licensed, distribution is impossible
-- Custom gmp found: /home/vanderson/mestrado/gmp/lib/libgmp.a
CMake Warning (dev) at src/solvers/yices/CMakeLists.txt:60 (target_link_libraries):
  Policy CMP0079 is not set: target_link_libraries allows use with targets in
  other directories.  Run "cmake --help-policy CMP0079" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  Target

    solvers

  is not created in this directory.  For compatibility with older versions of
  CMake, link library

    solveryices

  will be looked up in the directory in which the target was created rather
  than in this calling directory.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Testing enabled
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found version "1.71.0") found components: unit_test_framework
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vanderson/mestrado/esbmc/build

The output of cmake --build . command
[43/52] Generating ESBMC version ID
1+0 records in
1+0 records out
512 bytes copied, 8.4813e-05 s, 6.0 MB/s
[52/52] Linking CXX executable src/esbmc/esbmc
/usr/bin/ld: /home/vanderson/mestrado/clang9/lib/libLLVMSupport.a(Path.cpp.o): in function `llvm::sys::fs::expandTildeExpr(llvm::SmallVectorImpl<char>&)':
Path.cpp:(.text._ZN4llvm3sys2fsL15expandTildeExprERNS_15SmallVectorImplIcEE+0x133): warning: Using 'getpwnam' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /home/vanderson/mestrado/clang9/lib/libLLVMSupport.a(Path.cpp.o): in function `llvm::sys::path::native(llvm::SmallVectorImpl<char>&, llvm::sys::path::Style)':
Path.cpp:(.text._ZN4llvm3sys4path6nativeERNS_15SmallVectorImplIcEENS1_5StyleE+0x44f): warning: Using 'getpwuid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking

**Comment by rafaelsamenezes** [comment]
No errors there. It should be available at:

`/home/vanderson/mestrado/esbmc/build/src/esbmc/esbmc`

**Comment by vanderson-rocha** [comment]
@rafaelsamenezes 
I found the esbmc file in the folder mentioned. However, shouldn't it be (according to the building documentation) in  release folder?

**Comment by rafaelsamenezes** [comment]
Only after you install.

If you configured with ninja it should be:

`ninja install`

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: is our documentation is still consistent with the recent changes?

@vanderson-rocha has raised the point that our documentation gives wrong instructions. Or did I miss something?

**Comment by rafaelsamenezes** [comment]
> @rafaelsamenezes: is our documentation is still consistent with the recent changes?

Yes. The problem with the docs is that it assumes that the user will have a specific file tree:
- esbmc
- clang9
- ...

And also assumes that the user will build from `esbmc/build`. But if you follow the instructions it shouldn't have any problem.

> @vanderson-rocha has raised the point that our documentation gives wrong instructions. Or did I miss something?

I am unsure, we need to know if the `ninja install` command solved the issue. If the command is ok and the files are not there then we may have an issue with the docs.

**Comment by vanderson-rocha** [comment]
Then @rafaelsamenezes . I don't know if it's a problem on my side, but even when I run ninja install the esbmc file is not created (or replaced when it already exists) in release folder.

**Comment by rafaelsamenezes** [comment]
It should be in `$PWD/../../release` from where you run CMake. If it is not, could you attach your `CMakeCache.txt` file? It should be inside the folder in which you run CMake

**Comment by vanderson-rocha** [comment]
@rafaelsamenezes @lucasccordeiro. I found the problem. In the commit presented in https://github.com/esbmc/esbmc/commit/899ed69abaef2e662214749bbbb65ac9cf72ba4a#diff-396d0b75211f5646b2fa730fbdf80a40. there is a change in the paths used in cmake placing the executable in  release folder at same level as the esbmc folder. However when I read the documentation, I understand that release folder should be inside esbmc folder.

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: @vanderson-rocha has found a typo in the documentation. Could you please fix the location of the release folder?

**Comment by lucasccordeiro** [comment]
The fix seems to be simple. We just need to remove `..\`. Could you please check it?

**Comment by rafaelsamenezes** [comment]
Would putting:

> Once it is finished, ESBMC should be available at `../../release` from the build directory.

solve the issue?

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: @vanderson-rocha claims that the release is located at `../release` instead of `../../release`.

**Comment by vanderson-rocha** [comment]
@rafaelsamenezes , would solve. But, I understand that release folder must be inside the esbmc folder. My opinion.
